packages/test-runner/src/fixtures.ts

```diff
@@ -51,37 +52,39 @@ class Fixture {
   name: string;
   scope: Scope;
   fn: Function;
-  deps: string[];
-  usages: Set<string>;
+  deps: FixtureRegistration[];
+  usages: Set<FixtureRegistration>;
   hasGeneratorValue: boolean;
   value: any;
+  registration: FixtureRegistration;
   _teardownFenceCallback: (value?: unknown) => void;
   _tearDownComplete: Promise<void>;
   _setup = false;
   _teardown = false;
 
-  constructor(pool: FixturePool, name: string, scope: Scope, fn: any) {
+  constructor(pool: FixturePool, registration: FixtureRegistration) {
     this.pool = pool;
-    this.name = name;
-    this.scope = scope;
-    this.fn = fn;
-    this.deps = fixtureParameterNames(this.fn);
+    this.name = registration.name;
+    this.scope = registration.scope;
+    this.fn = registration.fn;
+    this.registration = registration;
+    this.deps = fixtureParameterNames(this.fn).map(name => name === this.name ? registration.super : getRegistration(name));
```

Comment:
This bit is a little sketchy. I am not saying it is wrong, but I am saying it could be more straightforward to be less error-prone. Instead of resolving the `FixtureRegistration` dependencies at an unknown fixture instantiation time, you should build a real single solid tree of FixtureRegistrations that would be based on the require order. You already do the right thing for `super`, but you should do that for `deps` as well. In the end of the day, `super` is just one of the `deps`, so `deps` should move from `Fixture` to `FixtureRegistration`. Then the reasoning about how dependency resolution works becomes so much easier!